README.md

```diff
@@ -0,0 +1,125 @@
+Kusto appender for Log4j.
+
+Motivation and usage.
+----------------------
+Log4j2 is widely used as logging tool. Kusto implementation
+is used in conjunction with RollingFileAppender with KustoStrategy.
+The key reason for using a strategy is to have redundancy in storage
+of logs and re-transmit the log files.
+
+To provide data transmission redundancy, the rolled over log files are
+transmitted to Kusto. Transmission of the files are attempted 3 times
+with a configured time window
+
+
+Adding appender to log4j.properties
+----------------------
+The key parameters for rolling file are as documented in
+the [Rolling file log4j configuration](https://logging.apache.org/log4j/2.x/manual/appenders.html#RollingFileAppender)
+
+- fileName: The file name where the log files will be written locally. This is a fully qualified path and not a
+  relative path (e.g C:/logs/logs.log)
+- filePattern: The rolled over file name with pattern. This is a fully qualified path and not a relative path (e.g C:
+  /logs/logs-%d{yyyy-MM-dd-hh-mm-ss}-%i.log)
+
+Configurations for using the Kusto log4j appender is as follows
+
+- KustoStrategy
+    - **clusterIngestUrl**: Ingest URL. Configured using environment variable **LOG4J2_ADX_INGEST_CLUSTER_URL**
+    - **appId**: Service principal application id. Configured using environment variable **LOG4J2_ADX_APP_ID**
+    - **appKey**: Service principal application secret. Configured using environment variable **LOG4J2_ADX_APP_KEY**
+    - **appTenant**: Tenant for the Service principal. Configured using environment variable **LOG4J2_ADX_TENANT_ID**
+    - **dbName**: Database name. Configured using environment variable **LOG4J2_ADX_DB_NAME**
+    - **tableName**: Table name for ingesting the logs
+    - **logTableMapping**: Mapping defined in the database to map the log data

```

Comment:
Mapping defined in the database to map the log data's fields to the table's columns.